core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/RulesDownloader.java

```diff
@@ -81,35 +95,179 @@ private void fetchRulesAndActiveRules(Rules.Builder rulesBuilder, Map<String, Ac
     }
   }
 
-  private static String getUrl(int page, int pageSize) {
+  private static String getUrl(int page, int pageSize, Version serverVersion) {
     StringBuilder builder = new StringBuilder(1024);
-    builder.append(RULES_SEARCH_URL);
+    builder.append(supportProtobuf(serverVersion) ? RULES_SEARCH_URL : RULES_SEARCH_URL_JSON);
     builder.append("&p=").append(page);
     builder.append("&ps=").append(pageSize);
     return builder.toString();
   }
 
-  private static SearchResponse loadFromStream(WsResponse response) {
-    try (InputStream is = response.contentStream()) {
-      return SearchResponse.parseFrom(is);
-    } catch (IOException e) {
-      throw new IllegalStateException("Failed to load rules", e);
+  private static SearchResponse loadFromStream(WsResponse response, Version serverVersion) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "loadFromStream" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)